src/kvstore/RocksEngine.cpp

```diff
@@ -106,6 +106,33 @@ RocksEngine::RocksEngine(GraphSpaceID spaceId,
     LOG(INFO) << "open rocksdb on " << path;
 }
 
+
+RocksEngine::RocksEngine(const std::string& dataPath,
+                         GraphSpaceID spaceId,
+                         int32_t vIdLen)
+    : KVEngine(spaceId), dataPath_(folly::stringPrintf("%s/nebula/%d", dataPath.c_str(), spaceId)) {
+    auto path = folly::stringPrintf("%s/data", dataPath_.c_str());
+    if (FileUtils::fileType(path.c_str()) == FileType::NOTEXIST) {
+        LOG(FATAL) << "Path " << path << " not exist";
+    }
+
+    if (FileUtils::fileType(path.c_str()) != FileType::DIRECTORY) {
+        LOG(FATAL) << path << " is not directory";
+    }
+
+    rocksdb::Options options;
+    rocksdb::DB* db = nullptr;
+    rocksdb::Status status = initRocksdbOptions(options, vIdLen);
+    CHECK(status.ok());
+
+    status = rocksdb::DB::OpenForReadOnly(options, path, &db);
```

Comment:
Readonly could just add a default parameter to previous ctor. We need to modify the `initRocksdbOptions()`, because two rocksdb may need to pass different configs, we use 3 gflags for all rocksdb for now.

Another option is directly pass a map, see https://github.com/facebook/rocksdb/wiki/Option-String-and-Option-Map. 

WDYT?

Reply:
Readonly rocksdb has the following differences:
1) path must exist
2) Read only does not require compaction_filter_factory
3) Use the OpenForReadOnly interface to initialize

As for the parameters, I think it’s okay, you can use flags to control

Reply:
yeah，will unify